deployment/tfhub/object-detection-base64.ipynb

```diff
@@ -0,0 +1,627 @@
+        "            }\n",
+        "        \n",
+        "        print(f\"Found {result['has_car']} object(s).\")\n",
+        "        return result\n",
+        "\n",
+        "    def run_detector(self, file, image_path):\n",
+        "        img = self.load_img(image_path)\n",
+        "        img = tf.image.convert_image_dtype(img, tf.float32)[tf.newaxis, ...]\n",
+        "        \n",
+        "        response = self.detector(img)\n",
+        "        result = self.filter_results(file, response)\n",
+        "        \n",
+        "        return result\n",
+        "\n",
+        "    def detect_objects(self, file, img_arr):\n",
+        "        start_time = time.time()\n",
+        "        image_path = self.handle_img(img_arr)\n",
+        "        result = self.run_detector(file, image_path)\n",
+        "        end_time = time.time()\n",
+        "\n",
+        "        print(f\"Inference time: {end_time - start_time}.\")\n",
+        "        return result\n",
+        "\n",
+        "    @verify_io\n",
+        "    def predict(self, data):\n",
+        "        file, img_arr = data\n",
+        "        result = self.detect_objects(file, img_arr)\n",
+        "\n",
+        "        return result"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5SvthV9gd4lk"
+      },
+      "source": [
+        "## 5. Model Register"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "UR7XtPNOHTgO",
+        "outputId": "43f4c6f0-a059-46bf-f89c-c541a3d34f94"
+      },
+      "outputs": [],
+      "source": [
+        "registered_model = client.get_or_create_registered_model(name='Object Detection - TF Hub')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "-RctZHJ8MMi5"
+      },
+      "outputs": [],
+      "source": [
+        "input = {\n",
+        "    \"file_name\": \"\",\n",
+        "    \"image_str\": \"\"\n",
+        "}\n",
+        "output = {\n",
+        "    \"file_name\": \"\",\n",
+        "    \"has_car\": False,\n",
+        "    \"score\": 0,\n",
+        "    \"ymin\": 0,\n",
+        "    \"xmin\": 0,\n",
+        "    \"ymax\": 0,\n",
+        "    \"xmax\": 0\n",
+        "}"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "vFlrCL2EWnJR",
+        "outputId": "5eaad4c4-d170-45c9-8d38-b2f037ed017f"
+      },
+      "outputs": [],
+      "source": [
+        "model = registered_model.create_standard_model(\n",
+        "    name = 'base64',\n",
+        "    model_cls = DetectObject,\n",
+        "    model_api = ModelAPI([input], [output]),\n",
```

Comment:
This is actually not the correct usage of `ModelAPI`. The first argument must match the _exact_ form of the input passed to the model `predict()`, so maybe `["file_name", "img_str"]`. The second argument must also match the output: Our model outputs `bboxes`, but the `output` variable here does not include `bboxes`.

Reply:
Yes, `output` was outdated. Changed to:
```
model_api = ModelAPI(
    [{'file_name': '', 'img': ''}],
    [{
        'file_name': '',
        'has_car': False,
        'score': 0,
        'bboxes': {
            'ymin': 0,
            'xmin': 0,
            'ymax': 0,
            'xmax': 0
        }
    }]
)
```